packages/ckeditor5-media-embed/src/mediaembedcommand.js

```diff
@@ -69,3 +64,29 @@ export default class MediaEmbedCommand extends Command {
 		}
 	}
 }
+
+// Checks if the table is allowed in the parent.
+//
+// @param {module:engine/model/selection~Selection|module:engine/model/documentselection~DocumentSelection} selection
+// @param {module:engine/model/schema~Schema} schema
+// @returns {Boolean}
+function isAllowedInParent( selection, model ) {
+	const insertPosition = findOptimalInsertionPosition( selection, model );
+	let parent = insertPosition.parent;
+
+	// The model.insertContent() will remove empty parent (unless it is a $root or a limit).
+	if ( parent.isEmpty && !model.schema.isLimit( parent ) ) {
+		parent = parent.parent;
+	}
+
+	return model.schema.checkChild( parent, 'media' );
+}
+
+// Checks if the media object is selected.
+//
+// @param {module:engine/model/selection~Selection|module:engine/model/documentselection~DocumentSelection} selection
+// @returns {Boolean}
+function isMediaSelected( selection ) {
+	const element = selection.getSelectedElement();
+	return element && element.name === 'media' || false;
```

Comment:
```suggestion
	return !!element && element.name === 'media';
```

is more natural and does not raise confusion regarding the order of operators.